### PR TITLE
Support for DTLS connection IDs

### PR DESCRIPTION
**Pull request by sergeuz**
### Problem

This PR enables the DTLS Connection ID extension (described in [this](https://tools.ietf.org/html/draft-ietf-tls-dtls-connection-id-09) draft RFC).

### Steps to Test

Verify that session resumption works as expected.

### References

- [ch55112]

### CODE REVIEW

File: communication/src/dtls_message_channel.cpp [hunk 1]
```diff
@@ -230,31 +258,16 @@ ProtocolError DTLSMessageChannel::init(
 	this->server_public = new uint8_t[server_public_len];
 	memcpy(this->server_public, server_public, server_public_len);
 	this->server_public_len = server_public_len;
+
+	// Records received from the server are not expected to contain a CID
+	mbedtls_ssl_conf_cid(&conf, 0 /* len */, MBEDTLS_SSL_UNEXPECTED_CID_IGNORE);
```
Comment (avtolstoy) [inline communication/src/dtls_message_channel.cpp:263]: Nit: `MBEDTLS_SSL_CID_IN_LEN_MAX` perhaps?

**Review by avtolstoy — APPROVED** [review]

**Comment by avtolstoy** [comment]
I rebased and fixed a couple of conflicts around persistent session.

**Comment by avtolstoy** [comment]
One thing to consider (as far as I understand DS side of this feature allows) is to remove server ip:port from the persistent storage and perform DNS resolution all the time to allow devices to resume the session on any DS instance. This should allow us to simplify DS deploys, have devices migrate to current DS instances quicker, increase maximum session times and improve connection times for sleepy devices in some cases.

File: communication/src/dtls_message_channel.cpp (hunk 1 above)
Comment (sergeuz) [reply communication/src/dtls_message_channel.cpp:264]: I'd prefer to use 0 here as it makes the intent clear that we don't expect a CID in server-originated messages. `MBEDTLS_SSL_CID_IN_LEN_MAX` defined as 0 in our config is more a memory optimization.

**Comment by sergeuz** [comment]
> I rebased and fixed a couple of conflicts around persistent session.

@avtolstoy Thanks!

**Pull request merged**